llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1116,9 +1116,8 @@ void VPWidenSelectEVLRecipe::execute(VPTransformState &State) {
   auto *InvarCond =
       isInvariantCond() ? State.get(getCond(), VPIteration(0, 0)) : nullptr;
 
+  // FIXME: Do we have a scenario where cond is a scalar?
```

Comment:
Better to add assertion here

Reply:
I might be missing something, but can't this happen when the condition is a loop invariant value? If so, this should have a. test.

Reply:
added assertion